core/authentication_api.php

```diff
@@ -209,7 +209,7 @@ function auth_auto_create_user( $p_username, $p_password ) {
 
 	if( $t_auto_create ) {
 		# attempt to create the user
-		$t_cookie_string = user_create( $p_username, md5( $p_password ) );
+		$t_cookie_string = user_create( $p_username, "" );
```

Comment:
Then we should not set it to a random password but a string that clearly indicates 'disabled password', such as '!' followed by a random hash. That way we don't need to generate a random password (and update the user account) on every login...